replica/replica.go

```diff
@@ -563,9 +563,12 @@ func (r *Replica) PrepareRemoveDisk(name string) ([]PrepareRemoveAction, error)
 		disk = GenerateSnapshotDiskName(name)
 		data, exists = r.diskData[disk]
 		if !exists {
-			return nil, fmt.Errorf("Can not find snapshot %v", disk)
+			return nil, nil
```

Comment:
how this is not error case? and moreover, why snapshot disk is generated here?

Reply:
If the disk is not present in the list, implies it has already been deleted. This check was valid when controller was triggering the delete. A snapshot might have been deleted from one replica but present in others. When replica deleted the snapshot by itself, this should never be hit.
GenerateSnapshotDiskName() just adds suffix and prefix to the snapshot name to get valid disk names.